javascript/ql/src/semmle/javascript/security/dataflow/ExceptionXss.qll

```diff
@@ -65,16 +125,20 @@ module ExceptionXss {
 
     override predicate isSanitizer(DataFlow::Node node) { node instanceof Xss::Shared::Sanitizer }
 
+    cached
```

Comment:
This doesn't immediately make sense to me: this predicate is only used in one query, so caching it is not useful. I suspect the performance benefits you were observing were due to `cached` preventing harmful magic. Could you try replacing this with `pragma[nomagic]`? If that has the same performance benefits, I would suggest using it instead since it more clearly communicates the intention.

Reply:
Adding `pragma[nomagic]` on `isAdditionalFlowStep` does not fix the performance. 

I also tried to add  `pragma[nomagic]` to `internalBlocks`, and that got me from a ~4X slowdown to a ~2X slowdown. 

I'll experiment some more.

Reply:
OK, there are a few other pragmas that `cached` implies, notably `pragma[noinline]`.

Reply:
After a whole lot of benchmarking I found that adding `pragma[noinline,nomagic]` on  `Configuration::internalBlocks` fixes the performance.   
Adding just `pragma[noinline]` or `pragma[nomagic]` is not enough, both are needed to avoid a performance regression on `oldj/SwitchHosts`. 

But with this change we will need a larger performance evaluation before we can merge.